### ISSUE TITLE
Socket.io does not guarantee that the second index in the request payload is the actuall ack.

### ISSUE THREAD

**Issue by FetzenRndy** [AndyGura/nestjs-socket-handlers-with-ack#2]
## The Problem

According to the [socket.io](https://socket.io/docs/client-api/#socketemiteventname-args-ack) documentation, it is not guaranteed that the second index of the request data is the ack.

For example you can emit a request like this: 

```ts
socket.emit("hello", { a: "b" }, "c", 45, console.log, 678, "Helloooo"); 
```

In this example the request data array will lock like this:

```ts
[{ a: "b" }, "c", 45, [function], 678, "Helloooo"]
```

When you just blindly access the second index ([here](https://github.com/AndyGura/nestjs-socket-handlers-with-ack/blob/d578c029a83acfdbe095da1a551049a6a6aa592b/src/models/socket-request.model.ts#L10)) you will end up invoking a string ([here](https://github.com/AndyGura/nestjs-socket-handlers-with-ack/blob/d578c029a83acfdbe095da1a551049a6a6aa592b/src/decorators/subscribe-message-with-ack.decorator.ts#L14)) since the "callback" property is the string "c"

According to the socket.io docs the last parameter **may be** a ack.
##### Relevant exerp from the socket.io docs.
```
Sending and getting data (acknowledgements)
Sometimes, you might want to get a callback when the client confirmed the message reception.

>> To do this, simply pass a function as the last parameter of .send or .emit <<
```

If you want me to send a pull request to fix this, let me know :) 

## How to fix ?

Just check if the last parameter of the payload is a function.

**Comment by AndyGura** [comment]
fixed in 787db33d66ef960e5dcab154629d11d47cf4193b

**Comment by AndyGura** [comment]
Fixed in version 1.1.0
